cmd/kubectl-gadget/deploy.go

```diff
@@ -64,293 +89,115 @@ func init() {
-              # syms_ vector will be empty and it will return false.
-              # As a consequence, no prefix will be found in
-              # get_syscall_prefix(), so a default prefix (_sys) will be
-              # returned.
-              # Sadly, this default prefix is not used by the running kernel,
-              # which instead uses: __x64_sys_
-              - SYSLOG
-
-              # traceloop gadget uses strace which in turns use ptrace()
-              # syscall.
-              # Within kernel code, ptrace() calls ptrace_attach() which in
-              # turns calls __ptrace_may_access() which calls ptrace_has_cap()
-              # where CAP_SYS_PTRACE is finally checked:
-              # https://elixir.bootlin.com/linux/v5.14.14/source/kernel/ptrace.c#L284
-              - SYS_PTRACE
-
-              # Needed by setrlimit in gadgettracermanager and by the traceloop
-              # gadget.
-              - SYS_RESOURCE
-
-              # Needed for gadgets that don't dumb the memory rlimit.
-              # (Currently only applies to BCC python-based gadgets)
-              - IPC_LOCK
-
-              # Needed by BCC python-based gadgets to load the kheaders module:
-              # https://github.com/iovisor/bcc/blob/v0.24.0/src/cc/frontends/clang/kbuild_helper.cc#L158
-              - SYS_MODULE
-
-              # Needed by gadgets that open a raw sock like dns and snisnoop
-              - NET_RAW
-        volumeMounts:
-        - name: host
-          mountPath: /host
-        - name: run
-          mountPath: /run
-        - name: modules
-          mountPath: /lib/modules
-        - name: debugfs
-          mountPath: /sys/kernel/debug
-        - name: cgroup
-          mountPath: /sys/fs/cgroup
-        - name: bpffs
-          mountPath: /sys/fs/bpf
-      tolerations:
-      - effect: NoSchedule
-        operator: Exists
-      - effect: NoExecute
-        operator: Exists
-      volumes:
-      - name: host
-        hostPath:
-          path: /
-      - name: run
-        hostPath:
-          path: /run
-      - name: cgroup
-        hostPath:
-          path: /sys/fs/cgroup
-      - name: modules
-        hostPath:
-          path: /lib/modules
-      - name: bpffs
-        hostPath:
-          path: /sys/fs/bpf
-      - name: debugfs
-        hostPath:
-          path: /sys/kernel/debug
-`
-
-type parameters struct {
-	Image               string
-	ImagePullPolicy     string
-	Version             string
-	HookMode            string
-	LivenessProbe       bool
-	FallbackPodInformer bool
+// parseK8sYaml parses a k8s YAML deployement file content and returns the
+// corresponding objects.
+// It was adapted from:
+// https://github.com/kubernetes/client-go/issues/193#issuecomment-363318588
+func parseK8sYaml(content string) ([]runtime.Object, error) {
+	acceptedK8sTypes := regexp.MustCompile(`(Namespace|ServiceAccount|Role|RoleBinding|ClusterRole|ClusterRoleBinding|DaemonSet|CustomResourceDefinition)`)
```

Comment:
nit: Do we really need to check this? In the end, we are parsing our own YAML. Additionally, every time we add a new resource, we will need to update this line.